airflow/kubernetes/kubernetes_helper_functions.py

```diff
@@ -29,15 +29,13 @@ def _strip_unsafe_kubernetes_special_chars(string: str) -> str:
     """
     Kubernetes only supports lowercase alphanumeric characters, "-" and "." in
     the pod name.
-    However, there are special rules about how "-" and "." can be used so let's
-    only keep
-    alphanumeric chars  see here for detail:
+    See here for detail:
     https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
 
     :param string: The requested Pod name
     :return: Pod name stripped of any unsafe characters
     """
-    return ''.join(ch.lower() for ch in list(string) if ch.isalnum())
+    return ''.join(ch.lower() for ch in list(string) if ch.isalnum() or ch in ['-'])
```

Comment:
It’s probably easier to do a `re.sub` instead.

Reply:
Can you elaborate more on why using an additional module would be more effective? I am fairly new to Python, so I am eager for knowledge.

Reply:
Not many reasons TBH, just it’s probably easier to understand IMO.

```python
return re.sub(r"[^-a-z0-9]+", string.lower())
```

It may also be easier to implement additional logic, such as avoiding `-` in the beginning of the string (which the logic you implement right now does not do).

Reply:
if we are to use regex, we should use a precompiled regex object. If we are not going to use regex here, then i think we should just do the comparison with `ch == '-'` without creating a new list in memory.